lib/MockXMLHttpRequest.js

```diff
@@ -139,61 +166,65 @@ MockXMLHttpRequest.prototype.overrideMimeType = function(mime) {
+          var response = MockXMLHttpRequest.handle(new MockRequest(self));
+          if (response && response instanceof MockResponse) {
+            var timeout = response.timeout();
+
+            if (timeout) {
+
+              //trigger a timeout event because the request timed out - wait for the timeout time because many libs like jquery and superagent use setTimeout to detect the error type
+              self._sendTimeout = setTimeout(function() {
+                self.readyState = MockXMLHttpRequest.STATE_DONE;
+                self.trigger('timeout');
+              }, typeof(timeout) === 'number' ? timeout : self.timeout+1);
+
+            } else {
+              //map the response to the XHR object
+              self.status             = response.status();
+              self._responseHeaders   = response.headers();
+              self.responseType       = 'text';
+              self.response           = response.body();
+              self.responseText       = response.body(); //TODO: detect an object and return JSON, detect XML and return XML
+              self.readyState         = MockXMLHttpRequest.STATE_DONE;
+
+              //trigger a load event because the request was received
+              self.trigger('load');
+
+            }
+          } else {
+
+            //trigger an error because the request was not handled
+            self.readyState = MockXMLHttpRequest.STATE_DONE;
+            self.trigger('error');
+
+          }
+      };
+
   this.data = data;
 
   self.readyState = MockXMLHttpRequest.STATE_LOADING;
 
-  self._sendTimeout = setTimeout(function() {
-
-    var response = MockXMLHttpRequest.handle(new MockRequest(self));
-
-    if (response && response instanceof MockResponse) {
-
-      var timeout = response.timeout();
-
-      if (timeout) {
-
-        //trigger a timeout event because the request timed out - wait for the timeout time because many libs like jquery and superagent use setTimeout to detect the error type
-        self._sendTimeout = setTimeout(function() {
-          self.readyState = MockXMLHttpRequest.STATE_DONE;
-          self.trigger('timeout');
-        }, typeof(timeout) === 'number' ? timeout : self.timeout+1);
-
-      } else {
-
-        //map the response to the XHR object
-        self.status             = response.status();
-        self._responseHeaders   = response.headers();
-        self.responseType       = 'text';
-        self.response           = response.body();
-        self.responseText       = response.body(); //TODO: detect an object and return JSON, detect XML and return XML
-        self.readyState         = MockXMLHttpRequest.STATE_DONE;
-
-        //trigger a load event because the request was received
-        self.trigger('load');
-
-      }
-
-    } else {
-
-      //trigger an error because the request was not handled
-      self.readyState = MockXMLHttpRequest.STATE_DONE;
-      self.trigger('error');
-
-    }
-
-  }, 0);
+  if(this.async) {
```

Comment:
When would we use the non-async version? shouldn't it always be async?

Will this be a breaking change for people using the library with a testing framework?